src/intersections/raysegment.jl

```diff
@@ -0,0 +1,75 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+#=
+The intersection type can be one of five types:
+
+1. intersect at one inner point (CrossingRaySegment -> Point)
+2. intersect at one corner point of segment xor origin of ray (MidTouchingRaySegment -> Point)
+3. intersects at one corner point of segment and origin of ray (CornerTouchingRaySegment -> Point)
+4. overlap at more than one point (OverlappingRaySegment -> Segment)
+5. do not overlap nor intersect (NoIntersection)
+=#
+function intersection(f, r1::Ray{N,T}, s1::Segment{N,T}) where {N,T}
+  a, b = r1(0), r1(1)
+  c, d = s1(0), s1(1)
+
+  λ₁, λ₂, r, rₐ = intersectparameters(a, b, c, d)
+
+  # not in same plane or parallel
+  if r ≠ rₐ
+    return @IT NoIntersection nothing f # CASE 5
+  # collinear
+  elseif r == rₐ == 1
+    r_c = sum((c - a) ./ r1.v)/N
+    r_d = sum((d - a) ./ r1.v)/N
+    r_c = isapprox(r_c, zero(T), atol=atol(T)) ? zero(T) : r_c
+    r_d = isapprox(r_d, zero(T), atol=atol(T)) ? zero(T) : r_d
```

Comment:
@dorn-gerhard please pay more attention to these issues that we keep raising in PRs. We should never access internal fields of structs, instead we should always use the public API. For instance, we never do r1.v in code because that is subject to change in the future.

A second issue that I raised already in other PRs is that we don't use underscores in our code style convention.

Reply:
Sorry, 
I wasn't aware of not using underscores. 
Thanks for clarifying conventions with accessing parts of ray. I will add a PR for correcting rays.jl as well.

Reply:
I cannot create a subscript c (always turns into a chi - not sure why). I therefore go for `rc`, `rd` as variable names